codegen/idris-codegen-c/Main.hs

```diff
@@ -44,9 +46,9 @@ c_main opts = do runIO setupBundledCC
 
 main :: IO ()
 main = do opts <- getOpts
-          if (null (inputs opts)) 
+          if (null (inputs opts))
              then showUsage
-             else runMain (c_main opts)
-
-
-
+             else if (not $ really opts)
+                     then do putStrLn "This is not what you may expect it is."
```

Comment:
I think that this string should be a bit more informative. Perhaps "This code generator is intended to be called by the Idris compiler. Please pass Idris the '--codegen' flag to choose a backend."